subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/git/GitConfigPlugin.java

```diff
@@ -0,0 +1,59 @@
+package org.shipkit.internal.gradle.git;
+
+import org.gradle.api.Action;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.shipkit.gradle.configuration.ShipkitConfiguration;
+import org.shipkit.gradle.exec.ShipkitExecTask;
+import org.shipkit.internal.gradle.configuration.ShipkitConfigurationPlugin;
+import org.shipkit.internal.gradle.exec.ExecCommandFactory;
+import org.shipkit.internal.gradle.util.TaskMaker;
+
+/**
+ * Tasks added by this plugin will set git repository option(s).
+ *
+ * The plugin applies following plugins:
+ *
+ * <ul>
+ *     <li>{@link ShipkitConfigurationPlugin}</li>
+ * </ul>
+ *
+ * and adds following tasks:
+ *
+ * <ul>
+ *     <li>
+ *         'setGitUserName' - sets generic user name so that CI server can commit code as neatly described robot,
+ *         uses value from {@link org.shipkit.gradle.configuration.ShipkitConfiguration.Git#getUser()}
+ *     </li>
+ *     <li>
+ *         'setGitUserEmail' - sets generic user email so that CI server can commit code as neatly described robot,
+ *         uses value from {@link org.shipkit.gradle.configuration.ShipkitConfiguration.Git#getEmail()}
+ *     </li>
+ * </ul>
+ */
+public class GitConfigPlugin implements Plugin<Project> {
```

Comment:
It is not related to the change but it would be nice to add unit tests for this plugin, to check if it can be applied and if it creates correct tasks. Or at least TODO in code if you don't have time for it at the moment.

Reply:
Yes, we should do so. I just added tests to check whether GitConfigPlugin and GitSetupPlugin can be applied.